src/cfnlint/rules/resources/ResourceSchema.py

```diff
@@ -0,0 +1,32 @@
+"""
+Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+SPDX-License-Identifier: MIT-0
+"""
+import os
+import re
+from glob import glob
+from jsonschema import validate, ValidationError
+from cfnlint.helpers import load_resource, REGEX_DYN_REF, PSEUDOPARAMS, FN_PREFIX, UNCONVERTED_SUFFIXES
+from cfnlint.rules import CloudFormationLintRule
+from cfnlint.rules import RuleMatch
+from cfnlint.data import ResourceSchemas
+
+class ResourceSchema(CloudFormationLintRule):
+    id = 'E3000'
+    shortdesc = 'Resource schema'
+    description = 'CloudFormation Registry resource schema validation'
+    source_url = 'https://github.com/aws-cloudformation/aws-cloudformation-resource-schema/'
+    tags = ['resources']
+
+    def match(self, cfn):
+        matches = []
+        for resource_schema in [os.path.basename(f) for f in glob('src/cfnlint/data/ResourceSchemas/*.json')]:
```

Comment:
so this would assume all the potential schemas are downloaded?

Reply:
wrote [`list-types`](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/list-types.html) / [`describe-type`](https://docs.aws.amazon.com/cli/latest/reference/cloudformation/describe-type.html) script to make it easy, but don't think it makes sense to call every runtime due to 
* credentials and networking requirements
* ability to provide additional resource schemas